core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/optimizers/FederationJoinOptimizer.java

```diff
@@ -370,6 +379,14 @@ else if (result != member) {
 			return result;
 		}
 
+		private boolean mayHaveStatement(RepositoryConnection conn, Resource subj, IRI pred, Value obj,
```

Comment:
Can this method be reused as it seems to be implemented both here and in EmptyPatternOptimizer?